src/Popover/elements.js

```diff
@@ -1,34 +1,113 @@
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
-export const PopOver = styled.div`
-position: relative;
+export const ARROW_SIZE = 8;
+export const CONTENT_PADDING = 1.8;
+export const MARGIN = 12;
 
-width: ${({ width }) => width};
+export const Arrow = styled.span`
+  color: ${({ theme: { palette } }) => palette.white};
 
-padding: 1.8rem;
+  text-shadow: ${({ placement }) =>
+    placement.startsWith('bottom')
+      ? '1px -4px 6px hsla(0,0%,0%,0.1)'
+      : placement.startsWith('left')
+      ? '1px -5px 5px hsla(0, 0%, 0%, 0.1)'
+      : placement.startsWith('right')
+      ? '1px -5px 5px hsla(0, 0%, 0%, 0.1)'
+      : '1px -5px 5px hsla(0, 0%, 0%, 0.1)'};
 
-border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+  position: absolute;
+  display: inline-block;
+  font-size: ${ARROW_SIZE * 2}px;
 
-box-shadow: 0 0.2rem 1.6rem 0 hsla(0, 0%, 0%, 0.1), 0 0.2rem 1.6rem 0 hsla(206, 23%, 69%, 0.1),
-  0 0 0 0.1rem hsl(207, 22%, 90%);
+  transform: ${({ placement }) =>
+    placement.startsWith('bottom')
+      ? 'rotate(0deg)'
+      : placement.startsWith('left')
+      ? 'rotate(90deg)'
+      : placement.startsWith('right')
+      ? 'rotate(-90deg)'
+      : 'rotate(180deg)'};
 
-background: ${({ theme: { palette } }) => palette.white};
+  ${({ placement }) =>
+    placement.startsWith('top') &&
+    css`
+      bottom: ${-CONTENT_PADDING + 0.5}rem;
+      left: calc(50% - ${ARROW_SIZE}px);
+      margin-bottom: 0;
+      margin-top: 0;
+    `}
 
-&::before {
-  content: "";
+  ${({ placement }) =>
+    placement.startsWith('bottom') &&
+    css`
+      top: ${-CONTENT_PADDING + 0.5}rem;
+      left: calc(50% - ${ARROW_SIZE}px);
+      margin-bottom: 0;
+      margin-top: 0;
+    `}
 
-  display:block;
+  ${({ placement }) =>
+    placement.startsWith('left') &&
+    css`
+      right: ${-CONTENT_PADDING + 0.8}rem;
+      top: calc(50% - ${ARROW_SIZE}px);
+      margin-left: 0;
+      margin-right: 0;
+    `}
 
-  position:absolute;
-  top: -0.7rem;
-  left: ${({ arrowPositionX }) => arrowPositionX};
-  z-index:-1;
+  ${({ placement }) =>
+    placement.startsWith('right') &&
+    css`
+      left: ${-CONTENT_PADDING + 0.8}rem;
+      top: calc(50% - ${ARROW_SIZE}px);
+      margin-left: 0;
+      margin-right: 0;
+    `}
+`;
+
+export const PopoverContentWrapper = styled.div`
+  background: ${({ theme: { palette } }) => palette.white};
+  border-radius: ${({ theme: { dimensions } }) => dimensions.radius};
+
+  box-shadow: 0 0.2rem 0.6rem 0 hsla(0, 0%, 0%, 0.1), 0 0.2rem 0.6rem 0 hsla(206, 23%, 69%, 0.1),
```

Comment:
You should use constant for color (hsl(207, 22%, 90%)). You can check with @arthurmonnet to find the right reusable color for the shadow